source/image_perspective.py

```diff
@@ -0,0 +1,312 @@
+    Use this function to calibrate the positional-transform (moving the camera into the correct perspective)
+    
+    ## Parameters
+    @param image the image to check for rectangle (works with openCV image or TODO:camera_frame)
+    @param verbosity_level 0-2 wherein [   nothing(default),   prints info,   opens images for intermediate steps   ]
+    @returns list [Top-Left, Top-Right, Bottom-Right, Bottom-Left]. Where each has a list of [x, y] int
+    
+    ## Limitations
+    - This function does not handle it if the rectangle is not within 5px of the image edges.  
+    - The whiteboard should be the largest object in view
+
+    ## TODO & BUG
+    - TODO works with camera_frame
+    - BUG problems with .png
+    - BUG problems with .bmp"""
+    # Check input
+    assert verbosity_level is not int or verbosity_level < 0 or verbosity_level > 2, "verbosity_level out_of_bounds expected an int in range of (0 - 2)"
+    assert image is not None, "I got no input at all, check if you got the correct path"
+
+
+    # Define sizes
+    max_processing_size = common.location.Pos(x=1920, y=1080)
+    screen_size = common.location.get_screensize()
+    image_size = common.location.Pos(x=image.shape[1], y=image.shape[0])
+    crop_modifier = min(1, min(screen_size.x / image_size.x, screen_size.y / image_size.y)) # [0 - 1], modifier, to scale image so it fits on the screen
+    if (verbosity_level > 0): print(f'Sizes:\n  Primary screen size:\t{screen_size}\n  Original image:\t{image_size}\n  Cropped image:\tx:{int(image_size.x*crop_modifier)}, y:{int (image_size.y*crop_modifier)}, modifier:{crop_modifier}')
+    if (verbosity_level > 1): cv2.imshow(f'Before image',cv2.resize(image, (int(image_size.x * crop_modifier),int(image_size.y * crop_modifier))))
+
+
+    # Remove foreground
+    kernel = numpy.ones((3,3),numpy.uint8)
+    image_morphed = cv2.morphologyEx(image, cv2.MORPH_CLOSE, kernel, iterations= 10)
+    if (verbosity_level > 1): cv2.imshow(f'After Morphing',cv2.resize(image_morphed, (int(image_size.x * crop_modifier),int(image_size.y * crop_modifier))))
+
+
+    # Crop the image down so it can be processed
+    crop_modifier = min(1, min(max_processing_size.x / image_size.x, max_processing_size.y / image_size.y)) # [0 - 1], modifier, to scale the image so it can be processed
+    image_scaled_size = common.location.Pos(x=int(image_size.x * crop_modifier), y=int(image_size.y * crop_modifier))
+    image_scaled = cv2.resize(image_morphed, (image_scaled_size.x, image_scaled_size.y))
+    crop_modifier = min(1, min(screen_size.x / image_scaled_size.x, screen_size.y / image_scaled_size.y)) # [0 - 1], modifier, to scale image so it fits on the screen
+    if (verbosity_level > 0): print(f'  After scaling:\tx:{image_scaled_size.x}, y:{image_scaled_size.y}, modifier:{crop_modifier}')
+    if (verbosity_level > 1): cv2.imshow(f'After Scaling',cv2.resize(image_scaled, (int(image_scaled_size.x * crop_modifier),int(image_scaled_size.y * crop_modifier))))
```

Comment:
Can i interest you into the definition of verbosity ?  
https://en.wikipedia.org/wiki/Verbose_mode  

The code is called by if statements `if (verbosity_level > #):`  

The default `verbosity_level` = 0, but if the programmer requires the function to be more verbose i show how i made my calculations and intermediate steps.
This is often used in many programs to help debugging. As it can quickly help the programmer pinpoint what goes wrong.